### ISSUE TITLE
bug of delete property not optional

### ISSUE THREAD

**Issue by anc95** [microsoft/TypeScript#40367]
**TypeScript Version:**  4.0.2

**Code**

```ts
const a: {a: string} = {a: 'name'};

delete a.a;

const b: {b: any} = {b: 'name'};

delete b.b;
```
**Expected behavior:**
should throw a lint error when on code of `delete b.b`
**Actual behavior:**
when the type be `{b: any}`, `delete b.b` will not throw an erorr
**Playground Link:** 
https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgLhgb0TaAnAlmA5gXxgF4U0ByMOAWwFMz8BuAKCYBMaAbGqG+AOjjMmoSLABGSZBPhgAnoRJSkFanUYt2XHjDF8xDIA

**Comment by RyanCavanaugh** [comment]
`undefined` is a legal value for `any`, so this is OK
